beaker-ts/src/notebook.ts

```diff
@@ -466,10 +592,34 @@ export class BeakerNotebook {
                 return new BeakerMarkdownCell(cell);
             }
         });
+
+        // attach children to parents: partition before attaching in one pass
+        // such that there's no dangling indices while removing elements from a list by iteration
+        //
+        // this is entirely a no-op if no cells are tagged as children; legacy notebooks of all parent cells are unaffected
+        const [parentCells, childCells]: [IBeakerCell[], IBeakerCell[]] = cellList.reduce(
```

Comment:
This algorithm feels overly complicated to me and likely hard to maintain.

I feel like this could be done with a single loop through the elements if we're not afraid of the list changing.

Something like:
```
let i = 0;
while(i < cellList.length) { // length should be checked every iteration so it's ok if it changes as long as i is in sync
  if (cellList[i].beaker_child_of !== undefined) {
   cellList.find(parent).children.push(cellList.splice(i, 1))
  }
  else {
    i++;
  }
}
```
Haven't tested and obviously somewhat pseudocode. but I _think_ something like that will work fine and be easier to maintain and less resource-heavy.